pkgs/games/runelite/default.nix

```
@@ -1,53 +1,65 @@
-{ stdenv, fetchurl, makeWrapper, jre, makeDesktopItem, lib }:
+{ stdenv, lib, pkgs, ... }:
 
 stdenv.mkDerivation rec {
   pname = "runelite";
-  version = "2.0.0";
+  version = "2.1.5";
 
-  src = fetchurl {
+  jar = pkgs.fetchurl {
     url = "https://github.com/runelite/launcher/releases/download/${version}/RuneLite.jar";
-    sha256 = "0q0x8g43ggkvp5fgnymgygx59xfhnyyrhpi6ha333gqg97rr0rvw";
+    sha256 = "4BX188QIjIFTxng2ktqlKn7AqQ9tdBcKWmgOj/5yd10=";
   };
 
-  icon = fetchurl {
+  icon = pkgs.fetchurl {
     url = "https://github.com/runelite/launcher/raw/${version}/appimage/runelite.png";
     sha256 = "04fcjm7p546gr82g0jbh497j7rnh70lrvas0k171bff4v3knrjw1";
   };
 
-  desktop = makeDesktopItem {
-    name = "RuneLite";
-    type = "Application";
-    exec = "runelite";
-    icon = icon;
-    comment = "Open source Old School RuneScape client";
-    terminal = "false";
-    desktopName = "RuneLite";
-    genericName = "Oldschool Runescape";
-    categories = "Game";
-    startupNotify = null;
+  # The `.so` files provided by these two jars aren't detected by RuneLite for some reason, so we have to provide them manually
+  jogl = pkgs.fetchurl {
+    url = "https://repo.runelite.net/net/runelite/jogl/jogl-all/2.4.0-rc-20200429/jogl-all-2.4.0-rc-20200429-natives-linux-amd64.jar";
+    sha256 = "S60qxyWY9RfyhLFygn/OxZFWnc8qVRtTFdWMbdu+2U0=";
+  };
+  gluegen = pkgs.fetchurl {
+    url = "https://repo.runelite.net/net/runelite/gluegen/gluegen-rt/2.4.0-rc-20200429/gluegen-rt-2.4.0-rc-20200429-natives-linux-amd64.jar";
+    sha256 = "eF8S5sQkJFDtW8rcVBKIyeyKm5Ze5rXK5r/yosZcHjU=";
   };
-
-  nativeBuildInputs = [ makeWrapper ];
-
-  # colon is bash form of no-op (do nothing)
   dontUnpack = true;
 
+  nativeBuildInputs = with pkgs; [ makeWrapper unzip ];
   installPhase = ''
     mkdir -p $out/share/runelite
     mkdir -p $out/share/applications
+    mkdir -p $out/natives
+
+    ${pkgs.unzip}/bin/unzip ${jogl}    'natives/*' -d $out
+    ${pkgs.unzip}/bin/unzip ${gluegen} 'natives/*' -d $out
```

```suggestion
    unzip ${jogl}    'natives/*' -d $out
    unzip ${gluegen} 'natives/*' -d $out
```

add to nativeBuildInputs.